src/components/legend_btn.tsx

```diff
@@ -16,14 +18,19 @@ function LEGEND_BTN(props: Props) {
     update_show_level_filter,
     show_level_filter,
   } = props;
-
+  const [, set_level_query_param] = useQueryParam(
+    "level",
+    mainQueryParamsConfig.level
+  );
```

Comment:
Will this not override the current query param on mount?

Reply:
nope, it just loads the setter with the config from mainQueryParamsConfig. 

```javascript
useQueryParam<T>(name: string, paramConfig: QueryParamConfig<T>, rawQuery?: ParsedQuery):
  [T | undefined, (newValue: T, updateType?: UrlUpdateType) => void]
```